doc/aaa/ldap/hld_ldap.md

```diff
@@ -0,0 +1,514 @@
+	global:{
+		"state": {{enabled/disabled}}
+		"bind_dn": {{ (string) }}
+		"bind_password": {{ ******** (string)}}
+		"bind_timeout": {{ (5 (duration_sec)) }}
+		"group_base_dn": {{ (string) }}
+		"group_member_attribute": {{member}}
+		" hostname_check ": {{False}}
+		" ldap_version": {{num}}
+		" user_base_dn": {{ou=users,dc=example,dc=com (string)}}
+		" login_name_attribute ": {{num}}
+		" ldap_port": {{num}}
+		" referrals": {{enable}}
+		" timeout": {{5 (duration_sec)}}
+	}
+}
+
+LDAP_SERVER:{
+	     <Sever-ip>:{
+			prio: {{prio-num}}
+	    }
+}
+AAA:{
+	Authentication:{
+			Failthought: {{“True”}}
+			Login :{{“local, ldap”}}
+	}
+}
+```
+		
+### Warmboot and Fastboot Design Impact  
+not relevant
+
+### Memory Consumption
+not relevant
+
+### Flows
+#### LDAP Authentication 
+The LDAP Pluggable Authentication Module (PAM) can be used to authenticate a CLI (SSH, or console) user to a Linux device like SONiC.
+More details:
+PAM - A module-based system for allowing service based authentication and accounting. Unlike NSS, you are not extending existing databases; PAM modules can use whatever logic they like, though shell logins still depend on the passwd and group databases of NSS. (you always need UID/GID lookups)
+
+#### LDAP BIND
+In order to use LDAP protocol is required to bind our switch to a LDAP server.
+That means, that after binding each client LDAP search to the Switch or SSH to the switch will be routed to the LDAP server.
+
+Flow of SSH login connection:
+Entry point  Command: ssh user_test@switch-133
+The user user_test will be searched in the LDAP server bound to the switch instead to the switch itself.
+The user authentication will be done by the LDAP server instead of the UNIX local authentication in the switch.
+
+How to do the binding configuration:
+The file that will contain all the info about binding is /etc/nslcd.conf.
+
+Those are the attributes required to set in order to bind client to server:
+
+binddn DN
+Specifies the distinguished name with which to bind to the directory server for lookups. The default is to bind anonymously.
+bindpw PASSWORD
+Specifies the credentials with which to bind. This option is only applicable when used with binddn above. If you set this option you should consider changing the permissions of the nslcd.conf file to only grant access to the root user.
+
+So, this file will contain all the LDAP configurations besides the login configuration that was described in the functional section above.
+Note: no restart is required when modifying PAM configuration. Only required to restart the NSLCD service after any modification in nslcd.conf file
+
+#### LDAP NSS
+LDAP can be used as an option in the Name Services Switch(NSS) configuration. The NSS configuration enables various programming APIs to use other sources than the default files (e.g., Use LDAP directory information instead of /etc/passwd for user and group information). User information includes uid, gid, and home directory.
```

Comment:
What is the rationale behind using NSL CD instead of SSSD  for ldap?  
There is a concern was the NSLCD does not support cache so it affects performance for scaling w.r.t many users. Is there any reason to go with NSLCD? 
Can you explore SSSD as well.

Reply:
NSLCD working smoothly, until today we haven't issues related to performances.
SSSD it's more complex to configure and use, so, for now, we go to the NSLCD direction.

But, it can be a future item to review. 
should be review:
 if the performance really enhances in our use cases and also if sssd makes the configuration even more complex to the customer from the UX (User experience).